internal/pkg/addons/storage.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package addons contains the service to manage addons.
+package addons
+
+import (
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/template"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+const (
+	dynamoDbAddonPath = "addons/ddb/cf.yml"
+	s3AddonPath       = "addons/s3/cf.yml"
+)
+
+type Storage struct {
+	ResourceName *string
+	Name         *string
+}
+
+type DynamoDB struct {
+	Storage
+
+	Attributes   []DDBAttribute
+	LSIs         []LocalSecondaryIndex
+	HasLSI       bool
+	SortKey      *string
+	PartitionKey *string
+
+	parser template.Parser
+}
+
+type S3 struct {
+	Storage
+	parser template.Parser
+}
+
+type StorageProps struct {
+	Name         string
+	ResourceName string
+}
+
+type S3AddonProps struct {
+	*StorageProps
+}
+
+type DynamoDbAddonProps struct {
```

Comment:
Maybe `DB` to conform with https://github.com/golang/go/wiki/CodeReviewComments#initialisms?